Cleanup includes in torch/csrc/jit/script/*

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #19885 Cleanup includes in c10/core/CPUAllocator.cpp.
* #19924 Cleanup includes in torch/csrc/*
* #19923 Cleanup includes in torch/csrc/autograd/*
* #19922 Cleanup includes in torch/csrc/jit/*
* **#19921 Cleanup includes in torch/csrc/jit/script/***

Differential Revision: [D15125017](https://our.internmc.facebook.com/intern/diff/D15125017)

Commit:
Cleanup includes in torch/csrc/jit/script/*

apaszke commented on torch/csrc/jit/script/tree_views.h:
```diff
@@ -1,10 +1,11 @@
+#include <torch/csrc/jit/script/tree.h>
 
 #include <functional>
+#include <iostream>
```
Why do we include `iostream` in a header?

ZolotukhinM replied:
There is a use of `std::cout` in `TreeView::dump()`. I guess we've been transitively including it through other headers, but I think it's better to explicitly include stuff we need (I didn't check for this very carefully, but fixed in places where I noticed that).

Krovatkin replied:
we could also consider starting tree_views.cpp and moving `dump` and a bunch of c-tors w/ `switch` stmts there 🤷‍♂

ZolotukhinM replied:
I agree. I'm actually planning on doing such cleanups (generally moving implementations out of header files) as well, but probably a bit later.

Commit:
Update on "Cleanup includes in torch/csrc/jit/script/*"

Cleanup includes in torch/csrc/jit/script/*

gh-metadata: pytorch pytorch 19921 gh/ZolotukhinM/75/head
